core/embed/rust/src/ui/model_mercury/flow/confirm_action.rs

```diff
@@ -240,53 +306,69 @@ fn create_flow(
 ) {
     let prompt_screen = prompt_screen.or_else(|| hold.then_some(title));
     let prompt_pages: usize = prompt_screen.is_some().into();
-
-    let (flow, page): (Result<SwipeFlow, Error>, &dyn FlowController) = if prompt_screen.is_some() {
-        (SwipeFlow::new(&ConfirmAction::Intro), &ConfirmAction::Intro)
-    } else {
-        (
-            SwipeFlow::new(&ConfirmActionSimple::Intro),
-            &ConfirmActionSimple::Intro,
-        )
+    let initial_page: &dyn FlowController = match (extra, prompt_screen.is_some()) {
+        (ConfirmActionExtra::Menu { .. }, false) => &ConfirmActionWithMenu::Action,
+        (ConfirmActionExtra::Menu { .. }, true) => &ConfirmActionWithMenuAndConfirmation::Action,
+        _ => &ConfirmAction::Action,
     };
 
-    (prompt_screen, prompt_pages, flow, page)
+    (
+        prompt_screen,
+        prompt_pages,
+        SwipeFlow::new(initial_page),
+        initial_page,
+    )
 }
 
 fn create_menu(
     flow: SwipeFlow,
-    menu: ConfirmActionMenu,
+    extra: ConfirmActionExtra,
     prompt_screen: Option<TString<'static>>,
 ) -> Result<SwipeFlow, Error> {
-    let mut menu_choices = VerticalMenu::empty().danger(
-        theme::ICON_CANCEL,
-        menu.verb_cancel.unwrap_or(TR::buttons__cancel.into()),
-    );
-
-    if menu.info {
-        menu_choices = menu_choices.item(
-            theme::ICON_CHEVRON_RIGHT,
-            menu.verb_info
-                .unwrap_or(TR::words__title_information.into()),
+    if let ConfirmActionExtra::Menu {
+        verb_cancel,
+        has_info,
+        verb_info,
+    } = extra
+    {
+        // NB: The Cancel menu item is always the first,
+        // because of the MENU_ITEM_CANCEL = 0.
+        // If we want the cancel item to be somewhere else,
+        // we would need to account for that and we could not use a constant.
+        let mut menu_choices = VerticalMenu::empty().danger(
+            theme::ICON_CANCEL,
+            verb_cancel.unwrap_or(TR::buttons__cancel.into()),
         );
-    }
 
-    let content_menu = Frame::left_aligned("".into(), menu_choices)
-        .with_cancel_button()
-        .with_swipe(Direction::Right, SwipeSettings::immediate());
+        if has_info {
```

Comment:
OK, something like this I guess: 1eff1d8e0e7660386e2475c7cdf99b63b35a9523

Now `new_confirm_blob` looks strange, but once I remove the `info` parameter completely, it will look great - basically the decision on whether to show info or not is done by passing `Some` or `None` in `verb_info`.

Reply:
Turned out slightly different actually, but anyway, we'll discuss this in the next PR, which does not cleanup on `confirm_blob`.